dask/compose.py

```diff
@@ -0,0 +1,66 @@
+from uuid import uuid4
+import operator
+from functools import partial, wraps
+
+from toolz import merge
+
+from .optimize import cull
+from .context import _globals
+from . import threaded
+
+
+def get_uuid_dasks(v):
+    if isinstance(v, Value):
+        return v._uuid, v._dasks
+    else:
+        return v, []
+
+
+def applyfunc(func, *args, **kwargs):
+    func = partial(func, **kwargs)
+    uuids, dsks = zip(*map(get_uuid_dasks, args))
+    uuid = uuid4().hex
+    dsks = sum(dsks, [{uuid: (func,) + uuids}])
+    return Value(uuid, dsks)
+
+
+def daskify(func):
+    """Wraps func so that it outputs a ``Value``"""
+    @wraps(func)
+    def _dfunc(*args, **kwargs):
+        return applyfunc(func, *args, **kwargs)
+    return _dfunc
+
+
+def get(dsk, keys, get=None, **kwargs):
+    """Specialized get function"""
+    get = get or _globals['get'] or threaded.get
+    dsk2 = cull(dsk, keys)
+    return get(dsk2, keys, **kwargs)
+
+
+class Value(object):
+    """Represents a value to be computed by dask.
+
+    Equivalent to the output from a single key in a dask graph.
+    """
+    def __init__(self, uuid, dasks):
+        self._uuid = uuid
+        self._dasks = dasks
+
```

Comment:
I would suggest that `Value` should have `dask` (computed on demand) and `name` as public API, like `dask.array`